From 3046e8f275c60f518da31acd054452d226798f54 Mon Sep 17 00:00:00 2001
From: Frederik Gladhorn <frederik.gladhorn@nokia.com>
Date: Fri, 23 Sep 2011 14:23:52 +0200
Subject: [PATCH] Fix accessible menu not returning child name.

When accessible name is set, QAccessibleMenu would always return
the menu's name, not that of the child action.

Task-Number: QTBUG-21578
Reviewed-by: Jan-Arve
(cherry picked from commit 3df9ac2f0719718ceb7385e46445447d95fd7857)
---
 src/plugins/accessible/widgets/qaccessiblemenu.cpp |    2 +-
 tests/auto/qaccessibility/tst_qaccessibility.cpp   |    6 ++++++
 2 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/src/plugins/accessible/widgets/qaccessiblemenu.cpp b/src/plugins/accessible/widgets/qaccessiblemenu.cpp
index 1454c7c..59e92c7 100644
--- a/src/plugins/accessible/widgets/qaccessiblemenu.cpp
+++ b/src/plugins/accessible/widgets/qaccessiblemenu.cpp
@@ -93,7 +93,7 @@ int QAccessibleMenu::childAt(int x, int y) const
 QString QAccessibleMenu::text(Text t, int child) const
 {
     QString tx = QAccessibleWidgetEx::text(t, child);
-    if (tx.size())
+    if (!child && tx.size())
         return tx;
 
     switch (t) {
